data_model/src/account.rs

```diff
@@ -1,45 +1,40 @@
 //! Structures, traits and impls related to `Account`s.
 
 #[cfg(not(feature = "std"))]
-use alloc::{
-    collections::{btree_map, btree_set},
-    format,
-    string::String,
-    vec::Vec,
-};
+use alloc::{collections::btree_map, format, string::String, vec::Vec};
 use core::{fmt, str::FromStr};
 #[cfg(feature = "std")]
-use std::collections::{btree_map, btree_set};
+use std::collections::btree_map;
 
-use iroha_data_primitives::small::{smallvec, SmallVec};
+use getset::{Getters, MutGetters, Setters};
+use iroha_data_primitives::small::SmallVec;
 use iroha_schema::IntoSchema;
 use parity_scale_codec::{Decode, Encode};
 use serde::{Deserialize, Serialize};
 
 #[cfg(feature = "roles")]
-use crate::role::Id as RoleId;
+use crate::role::{prelude::RoleId, RoleIds};
 use crate::{
-    asset::AssetsMap,
+    asset::{prelude::AssetId, AssetsMap},
     domain::prelude::*,
-    expression::{ContainsAny, ContextValue, EvaluatesTo, ExpressionBox, WhereBuilder},
+    expression::{ContainsAny, ContextValue, EvaluatesTo},
     metadata::Metadata,
-    permissions::PermissionToken,
-    Identifiable, Name, ParseError, PublicKey, Value,
+    permissions::{PermissionToken, Permissions},
+    prelude::Asset,
+    Identifiable, Name, ParseError, PublicKey,
 };
 
 /// `AccountsMap` provides an API to work with collection of key (`Id`) - value
 /// (`Account`) pairs.
-pub type AccountsMap = btree_map::BTreeMap<Id, Account>;
-
-/// Collection of [`PermissionToken`]s
-pub type Permissions = btree_set::BTreeSet<PermissionToken>;
+pub type AccountsMap = btree_map::BTreeMap<<Account as Identifiable>::Id, Account>;
 
 // The size of the array must be fixed. If we use more than `1` we
 // waste all of that space for all non-multisig accounts. If we
 // have 1 signatory per account, we keep the signature on the
 // stack. If we have more than 1, we keep everything on the
 // heap. Thanks to the union feature, we're not wasting `8Bytes`
 // of space, over `Vec`.
+// TODO: Should this be set rather than vec?
```

Comment:
I've looked at usage and I'm agree. Set structure can be more efficient. However I don't know if there will be some memory or (de-)serialization problem

Reply:
The memory usage is within a few bytes, and set can potentially have better performance

Reply:
The memory usage is within a few bytes, and set can potentially have better performance

Reply:
so I'm switching to set